app/src/features/apiClient/screens/PostmanImporterView/components/PostmanImporter/utils.ts

```diff
@@ -151,6 +157,107 @@ const processAuthorizationOptions = (item: PostmanAuth.Item | undefined, parentC
   return auth;
 };
 
+const getContentTypeForRawBody = (bodyType: string) => {
+  switch (bodyType) {
+    case "json":
+      return RequestContentType.JSON;
+    case "text":
+      return RequestContentType.RAW;
+    default:
+      return RequestContentType.RAW;
+  }
+};
+
+const addImplicitContentTypeHeader = (headers: KeyValuePair[], contentType: RequestContentType): KeyValuePair[] => {
+  const isContentTypeHeaderSet = headers.find((header: KeyValuePair) => header.key === "Content-Type");
+  if (!isContentTypeHeaderSet) {
+    return [
+      ...headers,
+      {
+        id: headers.length,
+        key: "Content-Type",
+        value: contentType,
+        isEnabled: true,
+      },
+    ];
+  }
+  return headers;
+};
+
+const processRawRequestBody = (raw: string, options: any, headers: KeyValuePair[]): RequestBodyProcessingResult => {
+  const contentType = getContentTypeForRawBody(options?.raw.language);
+  const updatedHeaders = raw.length ? addImplicitContentTypeHeader(headers, contentType) : headers;
+
+  return {
+    requestBody: raw,
+    contentType,
+    headers: updatedHeaders,
+  };
+};
+
+const processFormDataBody = (formdata: any[]): Omit<RequestBodyProcessingResult, "headers"> => {
+  return {
+    requestBody:
+      formdata?.map((formData: { key: string; value: string }) => ({
+        id: Date.now(),
+        key: formData.key,
+        value: formData.value,
+        isEnabled: true,
+      })) || [],
+    contentType: RequestContentType.FORM,
+  };
+};
+
+const processUrlEncodedBody = (urlencoded: any[], headers: KeyValuePair[]): RequestBodyProcessingResult => {
+  const contentType = RequestContentType.FORM;
+  const updatedHeaders = urlencoded.length ? addImplicitContentTypeHeader(headers, contentType) : headers;
+
+  return {
+    requestBody: urlencoded.map((data: { key: string; value: string }) => ({
+      id: Date.now() + Math.random(),
+      key: data?.key || "",
+      value: data?.value || "",
+      isEnabled: true,
+    })),
+    contentType,
+    headers: updatedHeaders,
+  };
+};
+
+const processRequestBody = (request: any): RequestBodyProcessingResult => {
+  if (!request.body) {
+    return {
+      requestBody: null,
+      contentType: RequestContentType.RAW,
+      headers: request.header || [],
+    };
+  }
+
+  const { mode, raw, formdata, options, urlencoded } = request.body;
+  const headers =
+    request.header?.map((header: KeyValuePair, index: number) => ({
+      id: index,
+      key: header.key,
+      value: header.value,
+      isEnabled: true,
+    })) ?? [];
+
+  switch (mode) {
+    case PostmanBodyMode.RAW:
+      return processRawRequestBody(raw, options, headers);
+    case PostmanBodyMode.FORMDATA:
+      return { ...processFormDataBody(formdata), headers };
+    case PostmanBodyMode.URL_ENCODED:
+      return processUrlEncodedBody(urlencoded, headers);
+    default:
+      return {
+        requestBody: null,
```

Comment:
> Raw mode with text body will be visible for now [_Ref_](https://github.com/requestly/requestly/pull/3015#discussion_r2097742166)

doesn't seem like it. Body just gets lost here (both in the data and in the UI). I tried with the current state of these changes and [this export file ](https://github.com/user-attachments/files/20347553/misc.postman_collection.json)

![CleanShot 2025-05-20 at 18 00 31@2x](https://github.com/user-attachments/assets/76428031-0033-4df1-ab29-6aa20723db44)